### PR TITLE
Allow option to generate index file

### PR DESCRIPTION
**Pull request by ruoniw**
Add the ability to generate an index file

### CODE REVIEW

**Review by Nickersoft — COMMENTED** [review]
It's getting there – could you also generate a changeset for this change using `bun changeset` and commit it?

File: packages/cli/src/commands/add/utils/index-file.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { writeFileSync } from "node:fs";
+import { readdir, rm, unlink } from "node:fs/promises";
+import { basename, extname, join } from "node:path";
+
+import { alphabetical } from "@atmx-org/registry/helpers/arrays/alphabetical.ts";
+
+function convertToFunctionName(fileName: string, isTypesRegistry: boolean): string {
```
[the remark this answers is not in the document]
Comment (ruoniw) [reply packages/cli/src/commands/add/utils/index-file.ts:7]: I do just coz I am iterating over all files in the directory (not just the new snippet that's added). I updated this function to just use pascal or camel util function tho.

File: packages/cli/src/commands/add/utils/index-file.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  const fileExtension = isTs ? '.ts' : '.js';
+  const validFiles = files.filter(
+    (file) => extname(file) === fileExtension && !file.includes("index"),
+  );
+
+  const exportExtension = isESM ? '.js' : '';
+  const exports = validFiles.map((file) => {
+    const fileName = basename(file, extname(file));
+    const functionName = convertToFunctionName(fileName, registryPath.includes('types'));
+    return `export { ${functionName} } from './${fileName}${exportExtension}';`;
```
[the remark this answers is not in the document]
Comment (ruoniw) [reply packages/cli/src/commands/add/utils/index-file.ts:30]: til

File: packages/cli/src/commands/add/utils/index-file.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  });
+
+  // Sort the exports alphabetically and join them into a string
+  const indexContent =
+    alphabetical(
+      exports,
+      (v) => v.match(/\{ (\w+) \}/)?.[1].toLowerCase() ?? v,
+    ).join("\n") + "\n";
+
+  writeFileSync(join(registryPath, "index" + fileExtension), indexContent);
```
[the remark this answers is not in the document]
Comment (ruoniw) [reply packages/cli/src/commands/add/utils/index-file.ts]: Updated! But why is async better than the sync version of the writefile fs function?

**Pull request merged**